script/publish.ts

```diff
@@ -151,9 +146,16 @@ function upload(assetName: string, assetPath: string) {
           hash.setEncoding('hex')
           const input = Fs.createReadStream(assetPath)
 
+          const uploadResult: IUploadResult = {
+            name: assetName,
+            url,
+            size: stats['size'],
+            sha: '',
+          }
```

Comment:
Couldn't you put this inside the `hash.on('finish'…` callback? That way you don't need to re-set the `uploadResult.sha`.

Reply:
Yep.